R/mod_export_count_table.R

```diff
@@ -0,0 +1,161 @@
+      ns <- session$ns
+
+
+      # Download modal
+      shiny::observeEvent(input[[EXP$ID$EXPORT_BUTTON]], {
+        # check validity of parameters
+
+
+        checkmate::assert(
+          checkmate::check_list(dataset()),
+          checkmate::check_subset(names(dataset()), choices = c("df", "meta")),
+          checkmate::check_data_frame(dataset()[["df"]]),
+          all(c("n_denominator", "hierarchy", "hier_lvl_col") %in% names(dataset()[["meta"]])),
+          combine = "and"
+        )
+
+        if (is.null(dataset())) {
+          shiny::showNotification(
+            "There is no dataset displayed currently. This may arise due to your filter choices or because the dataset is still loading.",
+            type = "message",
+            duration = NULL,
+            closeButton = TRUE,
+            id = NULL,
+            placement = "top-right"
+          )
+        } else {
+          shiny::showModal(
+            shiny::modalDialog(
+              shiny::tagList(
+                shiny::textInput(
+                  ns(EXP$ID$FILENAME_BOX),
+                  "Insert file name",
+                  value = "file_name"
+                ),
+                shiny::checkboxInput(
+                  inputId = ns(EXP$ID$DATAPROTECT_BOX),
+                  label = shiny::uiOutput(ns("label_id"), inline = TRUE),
+                  width = "100%"
+                )
+              ),
+              title = "Download box",
+              footer = list(shiny::fluidRow(
+                shiny::column(5, shinyjs::disabled(shiny::downloadButton(ns(EXP$ID$DOWNLOAD_BUTTON))), offset = 5),
+                shiny::column(1, shiny::modalButton("Cancel"))
+              )),
+              easyClose = TRUE
+            )
+          )
+        }
+      })
+
+
+      # Add/remove checkbox warning
+      checkbox_label <- shiny::eventReactive(input[[EXP$ID$DATAPROTECT_BOX]], {
+        if (input[[EXP$ID$DATAPROTECT_BOX]]) {
+          return(paste(EXP$MSG$DATAPROTECT_LABEL, intended_use_label))
+        } else {
+          return(shiny::tags$embed(
+            paste(EXP$MSG$DATAPROTECT_LABEL, intended_use_label),
+            shiny::br(shiny::tags$span(
+              style = paste("color:", "#CC3333"),
+              EXP$MSG$DATAPROTECT_WARN
+            ))
+          ))
+        }
+      })
+
+      # Update checkbox label separately (to allow testing of logic)
+      output$label_id <- shiny::renderUI(checkbox_label())
+
+
+      # Check if we should enable/disable download button
+      shiny::observe({
+        # Check if the inputs exist
+        if (!is.null(input[[EXP$ID$FILENAME_BOX]]) && !is.null(input[[EXP$ID$DATAPROTECT_BOX]])) {
+          # Check if the conditions are met
+          if (input[[EXP$ID$FILENAME_BOX]] != "" && input[[EXP$ID$DATAPROTECT_BOX]]) {
+            shinyjs::enable(EXP$ID$DOWNLOAD_BUTTON)
+          } else {
+            shinyjs::disable(EXP$ID$DOWNLOAD_BUTTON)
+          }
+        }
```

Comment:
This is mostly a stylistic suggestion. The only thing of value is the rejection of invalid destination file names. The rest, you can ignore:
```suggestion
        sanitized_fname <- fs::path_sanitize(input[[EXP$ID$FILENAME_BOX]])
        enable_button <- (nchar(sanitized_fname) > 0) && isTRUE(input[[EXP$ID$DATAPROTECT_BOX]])
        shinyjs::toggleState(id = EXP$ID$DOWNLOAD_BUTTON, condition = enable_button)
```